pkg/cloud/services/compute/instance.go

```diff
@@ -504,7 +510,13 @@ func createPort(is *Service, clusterName string, name string, net *infrav1.Netwo
 	if net.Subnet.ID != "" {
 		portCreateOpts.FixedIPs = []ports.IP{{SubnetID: net.Subnet.ID}}
 	}
+
+	mc := metrics.NewMetricPrometheusContext("port", "create")
+
 	newPort, err := ports.Create(is.networkClient, portCreateOpts).Extract()
+
+	mc.ObserveRequest(err)
+
 	if err != nil {
 		return ports.Port{}, fmt.Errorf("create port for server: %v", err)
 	}
```

Comment:
In line 534 and line 645 we could add a metric for `ports.Delete`.
In line 659 for `servers.Delete`.